packages/hooks/src/useTrackedEffect/index.ts

```diff
@@ -0,0 +1,22 @@
+import { useEffect, useRef, DependencyList } from 'react';
+
+const useTrackedEffect = (effect, deps?: DependencyList) => {
+  const initDeps = useRef<DependencyList>();
+  const diffTwoDeps = (deps1, deps2) => {
+    //Let's do a reference equality check on 2 dependency list.
+    //If deps1 is defined, we iterate over deps1 and do comparison on each element with equivalent element from deps2
+    //As this func is used only in this hook, we assume 2 deps always have same length.
```

Comment:
React `useEffect` allows the length of `deps` to be changed between the previous render and the next render. Maybe we should be consistent with `useEffect`?

Reply:
Nice suggestion but I think it's an edge case that we should exclude it out of this hook's scope. If we allow the length of deps to be changed, it'll kinda difficult to do a correct depedency list comparison and its output changes will be confusing to user.
For example, Dependency List have one more dependency in the middle of the list => Output indexes will be messed up.
My suggestion is we leave a comment or notice user about this case in our documentation.

Reply:
I think the current code is correct, But this comment is incorrect.

Although React deps allows changes, it does not focus on the variable length. like this

```ts
function depsAreSame(oldDeps: any[], deps: any[]): boolean {
  if (oldDeps === deps) return true;
  for (const i in oldDeps) {
    if (oldDeps[i] !== deps[i]) return false;
  }
  return true;
}
```